### PR TITLE
node_ops: add task manager module and base tasks classes

### PR DESCRIPTION
**Pull request by Deexie**
Add task manager node ops module and base classes for each node operation type.

### CODE REVIEW

**Comment by Deexie** [comment]
- s/booststrap_node_task_impl/bootstrap_node_task_impl
- delete unnecessary args from bootstrap_node_task_impl

File: node_ops/task_manager_module.cc
Comment (bhalevy) [inline node_ops/task_manager_module.cc:1]: the commit log says:
> Keep task manager node ops module in storage service.

but this is added under `node_ops/`.  I'm confused.

Comment (Deexie) [reply node_ops/task_manager_module.cc:1]: Oh, I've meant the literal object, will change

File: service/storage_service.cc [hunk 1]
```diff
@@ -3769,6 +3777,7 @@ future<> storage_service::replicate_to_all_cores(mutable_token_metadata_ptr tmpt
 }
 
 future<> storage_service::stop() {
+    co_await _task_manager_module->stop();
```
Comment (bhalevy) [inline service/storage_service.cc:3780]: Shouldn't we stop the t.m. module after aborting the node ops, as per the line below?
Or is stopping the module does that via the tasks interface, it should replace the below.

**Comment by Deexie** [comment]
- rebase
- change commitlog
- move node_ops::task_manager_module::stop call

**Comment by Deexie** [comment]
- rebase

**Review by bhalevy — APPROVED** [review]
LGTM.  @asias please review too.

But it does a bit pointless and out-of-context without adding the actual tasks.

**Comment by Deexie** [comment]
> But it does a bit pointless and out-of-context without adding the actual tasks.

Right, sorry, I wanted to split the code into possibly the smallest parts for easier and quicker review. I will open more meaningful patches from now on.

**Comment by Deexie** [comment]
> In general, I'd like to follow the linux kernel process of not adding unused code. It is sometimes ok to add a code in one patch and its usage in a following patch (and that can be documented in the commit log). But merging such a simple branch with no code using it seems wrong in this case, and it can be easily commited as part of a series that makes use of the added TM module.

Makes sense. I will open that together with task.

**Comment by Deexie** [comment]
Closing. Pushing the changes as a part of #15460.

**Pull request closed without merging**